src/features/offer/components/VenueSection/VenueSection.tsx

```diff
@@ -0,0 +1,114 @@
+import { useNavigation } from '@react-navigation/native'
+import React from 'react'
+import { View } from 'react-native'
+import { useQueryClient } from 'react-query'
+
+import { Coordinates, OfferVenueResponse, VenueResponse } from 'api/gen'
+import { UseNavigationType } from 'features/navigation/RootNavigator/types'
+import { formatFullAddressStartsWithPostalCode } from 'libs/address/useFormatFullAddress'
+import { analytics } from 'libs/firebase/analytics'
+import { useDistance } from 'libs/geolocation/hooks/useDistance'
+import { SeeItineraryButton } from 'libs/itinerary/components/SeeItineraryButton'
+import { getGoogleMapsItineraryUrl } from 'libs/itinerary/openGoogleMapsItinerary'
+import { QueryKeys } from 'libs/queryKeys'
+import { Separator } from 'ui/components/Separator'
+import { Spacer, Typo } from 'ui/theme'
+import { getHeadingAttrs } from 'ui/theme/typographyAttrs/getHeadingAttrs'
+
+import { VenueCard } from '../VenueCard/VenueCard'
+import { VenueDetails } from '../VenueDetails/VenueDetails'
+
+type Props = {
+  beforeNavigateToItinerary?: () => Promise<void> | void
+  venue: OfferVenueResponse | VenueResponse
+  locationCoordinates: Coordinates
+  title: string
+  showVenueBanner?: boolean | false
+}
+
+type PartialVenue = Pick<
+  VenueResponse,
+  'id' | 'venueTypeCode' | 'name' | 'description' | 'publicName'
+>
+
+const mergeVenueData =
+  (venue: PartialVenue) =>
+  (prevData: VenueResponse | undefined): VenueResponse => ({
+    id: venue.id,
+    name: venue.publicName || venue.name,
+    venueTypeCode: venue.venueTypeCode,
+    isVirtual: false,
+    description: venue.description,
+    accessibility: {},
+    contact: {},
+    ...(prevData || {}),
+  })
+
+export function VenueSection({
+  beforeNavigateToItinerary,
+  venue,
+  showVenueBanner,
+  locationCoordinates,
+  title,
+}: Props) {
+  const { navigate } = useNavigation<UseNavigationType>()
+  const queryClient = useQueryClient()
+  const { latitude: lat, longitude: lng } = locationCoordinates
+  const distanceToLocation = useDistance({ lat, lng })
+  const venueFullAddress = venue.address
+    ? formatFullAddressStartsWithPostalCode(venue.address, venue.postalCode, venue.city)
+    : ''
```

Comment:
Je ne pense pas qu'il soit nécessaire de faire ce check.
```suggestion
  const venueFullAddress = formatFullAddressStartsWithPostalCode(venue.address, venue.postalCode, venue.city)
```
puisqu'on voudrait peut-être afficher juste postalCode + city : `75000 Paris`.

Reply:
Yes tu as raison par contre il ne faut pas avoir le bouton "Voir l'itinéraire" si on n'a pas d'adresse précise. Je revois un petit peu les conditions